curriculum_learning.py

```
@@ -671,17 +674,25 @@ def _load_previous_stage_model(
+                    if self.rank == 0:
+                        print(
+                            f"⚠️  No previous stage checkpoint found, starting {current_stage} "
+                            f"from base pretrained LLM weights. ({e})"
+                        )
             return {
                 "stage": previous_stage,
                 "metrics": metrics,
                 "epoch": checkpoint.get("epoch", "?"),
                 "val_loss": checkpoint.get("val_loss", "?"),
             }
         except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)